### DISCUSSION TITLE
Difference between Slice and Array

### DISCUSSION THREAD

**Discussion by ib00** [biscuitlang/bl#161] [Q&A]
I thought I understood the idea of Slice (pointer + length), but then I got confused 
when I saw that memory can be allocated for a slice.

So, what's the difference between slice and array?

**Comment by travisdoor** [comment]
Hi, the main difference is that arrays are allocated on the stack every time, that's why we have to specify the size in compile time i.e. `foo: [2]s32`. To simplify things the interface of an array and slice is the same, we have `.len` and `.ptr`, but internally the size (or length) of an array is stored in the type itself and when `.len` is used on an array, it is implicitly replaced by constant number.
The slice on the other hand can point to any memory and `.len` + `.ptr` are explicitly stored in `struct { len: s64; ptr: *T; }`
